### PR TITLE
feat:  unify logic of checking update conditions

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

**Comment by Lukasz2891** [comment]
Have you checked the gelato-relayer parameters?

**Review by Lukasz2891 — DISMISSED** [review]

File: packages/on-chain-relayer/src/core/update-conditions/should-update.ts [hunk 1]
```diff
@@ -25,6 +31,19 @@ export const shouldUpdate = async (
     }
   }
 
+  for (const antiConditionCheck of Object.values(ANTI_CONDITIONS)) {
+    const { shouldNotUpdatePrice, message } = await antiConditionCheck(
+      context,
+      config
+    );
+    if (shouldNotUpdatePrice) {
+      shouldUpdatePrices = false;
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/on-chain-relayer/src/core/update-conditions/should-update.ts:40]: yep, but I think proposed solution by me is more straightforward and easier to understand

File: packages/on-chain-relayer/src/core/update-conditions/data-packages-timestamp.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { DataPackagesResponse } from "redstone-sdk";
+import { AntiConditionCheckFn, Context } from "../../types";
+
+export const checkIfDataPackageTimestampIsNewer: AntiConditionCheckFn = async (
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/on-chain-relayer/src/core/update-conditions/data-packages-timestamp.ts:4]: then old data package, I think this function name would be too long

File: packages/on-chain-relayer/src/core/update-conditions/data-packages-timestamp.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  context: Context
+): ReturnType<AntiConditionCheckFn> => {
+  const { dataPackages, lastUpdateTimestamp } = context;
+
+  const dataPackageTimestamp = await chooseDataPackagesTimestamp(dataPackages);
```
Comment (hatskier) [inline packages/on-chain-relayer/src/core/update-conditions/data-packages-timestamp.ts:9]: Why await?

**Comment by koonopek** [comment]
Merge two PRs in one, rename some function names.

**Review by hatskier — APPROVED** [review]

**Pull request merged**